app/Http/Controllers/MovieController.php

```diff
@@ -129,6 +129,10 @@ public function getActors(Request $request, Movie $movie)
 
     public function addActor(Movie $movie, Actor $actor)
     {
-
+        // Check if actor is in movie - if they are send error message!
+        if ($movie->actors->contains($actor)) return json_encode(array("message" => "This Actor already exists"));
```

Comment:
When creating an array, prefer to use the [ ] rather than the array function.

You can just return the array - Laravel will JSON encode it. Plus, it will also set the content-type HTTP header to application/json - which will be preferred by the client.

Client is currently returning a 200 OK in the case of an error. Consider moving the check into a validation rule (so we piggy back on the conventional validation mechanisms together with the responses etc.). Alternatively, consider not failing with an error, but instead only performing the operation if the actor is not already in the movie.

Reply:
Made changes RE you comments Nathan :)